src/main/java/net/mcreator/workspace/references/ObjectReference.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.workspace.references;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.METHOD) @Retention(RetentionPolicy.RUNTIME) public @interface ObjectReference {}

```

Comment:
Why isn't this called ModelReference, if it is used to mark Model references?

Reply:
I didn't intend to limit it to models exclusively.

Reply:
Well if method references textures, then you would likely use TextureReference, right?

Using universal object defies all we did so far with different annotation types for different reference types (texture, XML, model, ME, ...). "Object" isn't one of the references defined by MCreator